SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -0,0 +1,292 @@
+    return "denorm";
+  } else if constexpr (Val == init_val::inexact) {
+    return "inexact";
+  } else if constexpr (Val == init_val::ulp) {
+    return "ulp";
+  } else if constexpr (Val == init_val::ulp_half) {
+    return "ulp_half";
+  } else {
+    static_assert(Val != Val, "Unexpected enum value");
+  }
+}
+
+template <typename DataT, int NumElems, typename ContextT, init_val BaseVal,
+          init_val Step>
+class FillCtorTestDescription
+    : public TestDescription<DataT, NumElems, ContextT> {
+public:
+  FillCtorTestDescription(size_t index, DataT retrieved_val, DataT expected_val,
+                          const std::string &data_type)
+      : TestDescription<DataT, NumElems, ContextT>(index, retrieved_val,
+                                                   expected_val, data_type) {}
+
+  std::string to_string() const override {
+    std::string log_msg(
+        TestDescription<DataT, NumElems, ContextT>::to_string());
+
+    log_msg += ", with base value: " + init_val_to_string<BaseVal>();
+    log_msg += ", with step value: " + init_val_to_string<Step>();
+
+    return log_msg;
+  }
+};
+
+template <typename DataT, int NumElems, typename TestCaseT, typename BaseVal,
+          typename Step>
+class run_test {
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    static_assert(std::is_same_v<typename BaseVal::value_type, init_val>,
+                  "BaseVal template parameter should be init_val type.");
+    static_assert(std::is_same_v<typename Step::value_type, init_val>,
+                  "Step template parameter should be init_val type.");
+
+    shared_vector<DataT> result(NumElems, shared_allocator<DataT>(queue));
+
+    const auto base_value = get_value<DataT, BaseVal::value>();
+    const auto step_value = get_value<DataT, Step::value>(base_value);
+
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *const out = result.data();
+
+      cgh.single_task<kernel_for_fill<DataT, NumElems, TestCaseT,
+                                      BaseVal::value, Step::value>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            TestCaseT::template call_simd_ctor<DataT, NumElems>(
+                base_value, step_value, out);
+          });
+    });
+    queue.wait_and_throw();
+    bool passed = true;
+    if (!are_bitwise_equal(result[0], base_value)) {
+      // Verify the base value was passed as-is
+      passed = fail_test(0, result[0], base_value, data_type);
+    }
+
+    DataT expected_value = base_value;
+    // Verify the step value works as expected being passed to the fill
+    // constructor.
+    for (size_t i = 1; i < result.size(); ++i) {
+      if constexpr (BaseVal::value == init_val::nan ||
+                    Step::value == init_val::nan) {
+
+        if (!std::isnan(result[i])) {
+          passed = false;
+
+          // TODO: Make ITestDescription architecture more flexible
+          std::string log_msg = "Failed for simd<";
```

Comment:
```suggestion
          // TODO: Make ITestDescription architecture more flexible
          // We are assuming that the NaN opcode may differ
          std::string log_msg = "Failed for simd<";
```

Reply:
this comment was applied in this commit: 292118ac7aa44deec887e1fb9b977fa49e636820